### DISCUSSION TITLE
How to search for spaces?

### DISCUSSION THREAD

**Discussion by luckman212** [BurntSushi/ripgrep#2042] [Q&A] [answered]
Sorry for this rudimentary question, but I am wondering if this is correct to search for `"some pattern"` (words separated by a space):

```shell
rg 'some pattern'
```
because my instinct was to escape the space, like this
```shell
rg 'some\ pattern'
```
which results in
```
regex parse error:
    some\ pattern
        ^^
error: unrecognized escape sequence
```
similarly, how to search for a line ending with 2 spaces?
```shell
rg '  $'
```
not?
```shell
rg '\ \ $'
```
any way to invoke `rg` in a way that the syntax is more aligned with standard unix grep/sed etc?
thank you

**Comment by BurntSushi** [comment, marked as answer]
Why are you trying to escape spaces? Literal spaces work just fine. What isn't working about them?

> more aligned with standard unix grep/sed etc

POSIX support is a non-goal. If you want POSIX then use POSIX compatible tooling.

**Comment by luckman212** [comment]
Ok @BurntSushi thank you. Makes sense. Sorry for the uninformed question.

I guess I picked up a habit of escaping spaces somewhere along my way, maybe when working with unquoted strings ... `rg foo\ bar` instead of `rg 'foo bar'` but I understand that is apples & oranges

**Reply by BurntSushi** [reply]
Well, `rg foo\ bar` works. It's just `rg 'foo\ bar'` that doesn't work.

FWIW, it's likely that escaping spaces will ultimately be allowed. It's just that it's a property of the underlying regex engine, not ripgrep.

**Comment by gvanem** [comment]
I have a similar issue. Searching for `fprintf(` causes a parse error (`error: unclosed group`). 

I'd like to search w/o any regexp-syntax completely.  But I fail to see there is one.

**Reply by BurntSushi** [reply]
That's what the `-F/--fixed-strings` flag is for. It's been standard in grep tooling for decades:

```
       -F, --fixed-strings
           Treat the pattern as a literal string instead of a regular expression. When this flag is used,
           special regular expression meta characters such as .(){}*+ do not need to be escaped.
```

**Reply by BurntSushi** [reply]
And to be clear, that isn't too similar to this issue. This issue is about being able to escape something that doesn't actually need escaping. Your issue is the opposite: you're trying to search for a literal `(` without escaping it, despite the fact that `(` is a regex meta character.

**Reply by gvanem** [reply]
@BurntSushi Excellent! I did not know about this `-F` option.

**Comment by luckman212** [comment]
@gvanem You can use either of these...

```
rg --fixed-strings 'fprintf('
```
or
```
rg 'fprintf\('
```

**Reply by BurntSushi** [reply]
Note that `-F` is the abbreviated form of `--fixed-strings`.
